cloudformation/transformation.yaml

```diff
@@ -0,0 +1,120 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+
+Resources:
+  TransformationBucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      BucketName: !Sub ${AWS::StackName}-transformation
+      AccessControl: Private
+      PublicAccessBlockConfiguration:
+        BlockPublicAcls: true
+        BlockPublicPolicy: true
+        IgnorePublicAcls: true
+        RestrictPublicBuckets: true
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+          - ServerSideEncryptionByDefault:
+              SSEAlgorithm: aws:kms
+              KMSMasterKeyID: !GetAtt KmsKey.Arn
+      VersioningConfiguration:
+        Status: Enabled
+      LoggingConfiguration:
+        DestinationBucketName: !Ref GlobalLogBucket
+        LogFilePrefix: transformation-bucket/log
+      NotificationConfiguration:
+        LambdaConfigurations:
+          - Event: s3:ObjectCreated:*
+            Filter:
+              S3Key:
+                Rules:
+                  - Name: suffix
+                    Value: '.csv'
+            Function: !GetAtt TransformationFunction.Arn
+
+  TransformationFunction:
+    # checkov:skip=CKV_AWS_116: DLQ not needed for lambda driven by SQS
```

Comment:
It does not seem to be the case that events triggering this function come from an SQS-queue (with its capability to defer to DLQ in case of problems). So we would either need a DLQ for this function, or an SQS-queue (with DLQ), where the S3-events go.

Reply:
Seeing that there is an alarm associated with the function, this might not be necessary depending on how this functionality is supposed to be used. If we don't need to keep track of failures then it's fine as is.